src/rest_api.py

```diff
@@ -129,13 +129,21 @@ def report():
                 "scanned_at": task_result.get("scanned_at"),
                 "dependencies": task_result.get("dependencies")
             })
+
+            if task_result.get('lock_file_absent'):
+                response.update({
+                    "lock_file_absent": task_result.get("lock_file_absent"),
+                    "message": task_result.get("message")
+                })
+                return flask.jsonify(response), 400
+
             return flask.jsonify(response), 200
         else:
             response.update({
                 "status": "failure",
                 "message": "Failed to retrieve scan report"
             })
-            return flask.jsonify(response), 404
+            return flask.jsonify(response), 500
```

Comment:
@samuzzal-choudhury I changed this to 500 status code, since it's more of an internal server error rather than report not available.

Reply:
@abs51295 ack!